contract/src/main/resources/swagger/kafbat-ui-api.yaml

```diff
@@ -1048,6 +1048,32 @@ paths:
         200:
           description: OK
 
+  /api/clusters/{clusterName}/consumer-groups/{id}/topics/{topicName}:
+    delete:
+      tags:
+        - Consumer Groups
+      summary: delete consumer group offsets
```

Comment:
In API spec it's 'delete' offsets, in java code it's 'reset' (`.consumerGroupActions(groupId, RESET_OFFSETS)`). I suggest let's decide on one thing. Since we already have 'reset offset' in consumers, I suggest keeping 'reset' in favor of 'delete' here as well. What do you think?

Reply:
I named it 'delete' since the kafka-cli uses 'delete-offsets'. 
Also, because this feature deletes the offsets of the consumer group rather than deleting the consumer group itself, I thought the ```RESET_OFFSETS``` was more appropriate than the ```DELETE``` for offset deleting.

And since the "reset consumer group offsets" feature already exists, I think distinguishing it as "delete" would be better. Another option could be splitting the actions, something like ```.consumerGroupActions(groupId, DELETE_OFFSETS)```, just like we do with ```.consumerGroupActions(groupId, RESET_OFFSETS).``` What do you think?

Reply:
Thanks, these are good points.

Regarding the permission naming, there are a few things I would consider when a question of (not) introducing a new permission arises:
- Consider existing 'sibling' permissions: If we tie new functionality to existing permission, how bad can it be for the end users? What's the chance that somebody wouldn't want to automatically assign permission for such an action to folks who already have the aforementioned permission granted to them? In this particular case, I believe, having permission to reset offsets (to zero) is kind of close to being able to delete offsets altogether, so it seems pretty safe to have a single permission for both actions (and we can always add a new one if we decide otherwise).
- Granularity: We don't want to introduce a new permission unless it's necessary, otherwise it's just an extra overhead with adding more overwhelming choices to grant these new permissions to people.

Given these, I believe using `RESET_OFFSETS` for deleting offsets is okay, what is your opinion on this, @p-eye?

cc @kafbat/backend @germanosin

Reply:
I completely agree with your points !